### ISSUE TITLE
pushObject.on('notification') not working at all 

### ISSUE THREAD

**Issue by prof-jagpreet** [phonegap/phonegap-plugin-push#2724]
### Expected Behaviour

1.When ever i receive a push notification in foreground an alert box should appear. Its not happening. 
2. If application is in background then my Complaints page should open because i am sending the id in data field.

### Actual Behaviour

Nothing happens whether i am in foreground or background.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Adorid : 8.1
#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
Redmi Note 6 Pro
#### Cordova CLI version and cordova platform version

    cordova --version                                   8.0.0
    cordova platform version android          7.1.4

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push   2.2.3

#### Sample Code that illustrates the problem
This is my code

 this.pushObject.on('notification').subscribe((data: any) => {
        console.log('message -> ' + data.message);
        this.mydata ="received";
        //if user using app and push notification comes
        if (data.additionalData.foreground) {
          this.alerter.create({
            title: data.ntitle,
            message: data.nmessage,
            buttons: [
              {
                text: 'View',
                handler: () => {
                  this.navCtrl.push(ComplaintDetailsPage, { complaintId: data.id * 1 });
                }
              },
              {
                text: 'Hide',
                role: 'cancel',
                handler: () => {
                  //console.log('Cancel clicked');
                }
              }
              
            ]
          }).present();
        } else {
          //if user NOT using app and push notification comes
          //TODO: Your logic on click of push notification directly
          this.navCtrl.push(ComplaintDetailsPage, { complaintId: data.id * 1 });
          
        }
      });

#### Logs taken while reproducing problem

This is happening when i update my ionic api and all plugins. Other plugins working fine. Issue is only with Push Notifications

**Comment by rychter** [comment]
Could you add Sample Push Data Payload? 

Try: 

{
  priority: 'high', 
  data: {
    count: 5, 
    sound: 'default', 
    notId: 1, 
    'content-available': '1', 
    custom_key1: 'value1',
    custom_key2: 'value2',
    title: 'A short string desk purpose of the notification',
    message: 'The text of the alert message'
  }
}


If you have local notification plugin remove it.

**Comment by rychter** [comment]
Also this may help: https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#chinese-android-phones

**Comment by prof-jagpreet** [comment]
> Could you add Sample Push Data Payload?
> 
> Try:
> 
> {
> priority: 'high',
> data: {
> count: 5,
> sound: 'default',
> notId: 1,
> 'content-available': '1',
> custom_key1: 'value1',
> custom_key2: 'value2',
> title: 'A short string desk purpose of the notification',
> message: 'The text of the alert message'
> }
> }
> 
> If you have local notification plugin, remove it.

I tried your payload.. but same issue... it looks like notifications are working but methods in the plugin are not working.

**Comment by rychter** [comment]
I had the same issue, then: 
1) I started using payload with data
2) I removed local notification plugin 
and it work ok. 

You may also check the link that I have posted about Chinese phones.

**Comment by CSobol** [comment]
Yet another problem that would be fixed if the maintainers were to accept #2369

**Comment by bhaskar8088** [comment]
I set content-available = 1 but when I click system notification, it opens app but notification event not getting fired.

Not sure whats the issue. Its working for everyone as of now? 

My phonegap plugin push version - 2.1.3

I am sending notification using https://github.com/olucurious/PyFCM/

When I check logs for payload, its sending like this:

`{"android":{"priority":"high"},"data":{"content-available":1,"image":"some image link","abc": "some value"},"notification":{"body":"some content", "title":"some title"},"registration_ids": ["one", "two"]}`

**Comment by angelique360** [comment]
Hi everyone 

I tested this plugin and i have a lot of problems but at the end i can send and recive push notification on ios and android.

First i have to send different payload for android and ios because if i only send one  there are problems.

IOS Example:

`{ 
	"to" : "/topics/ios-evento_29901",
	"content_available":false,
	"notification": {
    "body" : "Mensaje en 3RA RONDA - ADMIN",
    "content_available": 1
		,
		"sound": "default",
		"badge":"1",
		"click_action":"FCM_PLUGIN_ACTIVITY"
  },
  "data": {
		"content_available": 1,
		"message": "Mensaje en 3RA RONDA - ADMIN",
		"eventoId":"29901",
		"ejecucionId":"23916",
		"tipo":"COMENTARIO",
		"usuarioId":"1",
		"fecha":"1550854587380"
	},
  "priority" : "high"
}` 


Android example:

`{
  "to": "/topics/evento_19986",
  "data": {
    "message": "This is a Firebase Cloud Messaging Topic Message! 3",
    "eventoId":"19986",
    "ejecucionId":"17052",
    "tipo":"COMENTARIO",
    "usuarioId":"0",
    "fecha":"2018-11-09 13:52:16.787"
  }
}
`
In Android works like a charm but in IOS not
I have Xcode 10
So after many intents I was able to recive Push, I changed WorkSpace Settings -> Build System to Legacy Build System

I tested on IOS Device, 

iOS Version 9 Works!
IOS Version (10-11-12) the event on('notification') not fired 

i expect somebody can help me with this bug!!

(sorry for my English)

**Comment by CSobol** [comment]
You are correct that this will work, however it's a silly workaround that should not be needed to use FCM for cross platform notifications.

The true fix is in PR #2369

**Comment by duca14036** [comment]
solution for me was to use payload in this format
{
  "to" : "bk3RNwTe3H0:CI2k_HHwgIpoDKCIZvvDMExUdFQ3P1...",
  "notification": {},
  "data": {},
  "content_available": true  
}
and add code from old version which trigger notification event when app is in foreground
file  from IOS project, Plugins/AppDelegate+notification.m
in func didReceiveRemoteNotification
added this code from 'else':
``` 
if (application.applicationState != UIApplicationStateActive) {
...
} else {
       NSLog(@"app active");
        PushPlugin *pushHandler = [self getCommandInstance:@"PushNotification"];
        pushHandler.notificationMessage = userInfo;
        pushHandler.isInline = YES;
        [pushHandler notificationReceived];
        
        completionHandler(UIBackgroundFetchResultNewData);
}

**Comment by lamuertepeluda** [comment]
can this be related with #2738 ?
@CSobol I also tried your version of the plugin, but no luck 😞 
the `on('notification')` just does not get fired...

**Comment by pabloleone** [comment]
This PR solves the issue https://github.com/phonegap/phonegap-plugin-push/pull/2369

**Comment by bunditng** [comment]
> Could you add Sample Push Data Payload?
> 
> Try:
> 
> {
> priority: 'high',
> data: {
> count: 5,
> sound: 'default',
> notId: 1,
> 'content-available': '1',
> custom_key1: 'value1',
> custom_key2: 'value2',
> title: 'A short string desk purpose of the notification',
> message: 'The text of the alert message'
> }
> }
> 
> If you have local notification plugin, remove it.

thank you **@rychter** 
think it because by **local-notificaions** plugin. after remove it, work properly
